### PR TITLE
Remove v2 resharding fields

### LINKED ISSUE #10378: Task: delete unused sharding_column_name and sharding_column_type
### Question

Keyspace protobuf definition still contains sharding_column_name and sharding_column_type. These are from v2 resharding which predates vindex based resharding. These fields can now be removed.

### PR DESCRIPTION
**Pull request by arvind-murty**
## Description

`sharding_column_name` and `sharding_column_type` are no longer used by resharding, so they can be deleted.

## Related Issue(s)

Fixes #10378 

## Checklist

-   [ ] "Backport me!" label has been added if this change should be backported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

**Comment by deepthi** [comment]
@rohit-nayak-ps vtctld_web_test is failing on this PR. Do we need to regenerate the rice-box.go for these changes? Is that still possible?
```
2022-06-02T00:53:12.6438684Z     vtctld_web_main_test.go:481: 
2022-06-02T00:53:12.6439138Z         	Error Trace:	vtctld_web_main_test.go:481
2022-06-02T00:53:12.6439701Z         	            				vtctld_web_test.go:127
2022-06-02T00:53:12.6440023Z         	Error:      	elements differ
2022-06-02T00:53:12.6440310Z         	            	
2022-06-02T00:53:12.6440745Z         	            	extra elements in list B:
2022-06-02T00:53:12.6441187Z         	            	([]interface {}) (len=2) {
2022-06-02T00:53:12.6442053Z         	            	 (string) (len=29) "-sharding_column_name=test_id",
2022-06-02T00:53:12.6442783Z         	            	 (string) (len=28) "-sharding_column_type=UINT64"
2022-06-02T00:53:12.6443104Z         	            	}
2022-06-02T00:53:12.6443368Z         	            	
2022-06-02T00:53:12.6443622Z         	            	
2022-06-02T00:53:12.6443910Z         	            	listA:
2022-06-02T00:53:12.6444308Z         	            	([]string) (len=3) {
2022-06-02T00:53:12.6444834Z         	            	 (string) (len=14) "CreateKeyspace",
2022-06-02T00:53:12.6445445Z         	            	 (string) (len=12) "-force=false",
2022-06-02T00:53:12.6445953Z         	            	 (string) (len=14) "test_keyspace3"
2022-06-02T00:53:12.6446254Z         	            	}
2022-06-02T00:53:12.6446515Z         	            	
2022-06-02T00:53:12.6447017Z         	            	
2022-06-02T00:53:12.6447309Z         	            	listB:
2022-06-02T00:53:12.6447705Z         	            	([]string) (len=5) {
2022-06-02T00:53:12.6448204Z         	            	 (string) (len=14) "CreateKeyspace",
2022-06-02T00:53:12.6448909Z         	            	 (string) (len=29) "-sharding_column_name=test_id",
2022-06-02T00:53:12.6449626Z         	            	 (string) (len=28) "-sharding_column_type=UINT64",
2022-06-02T00:53:12.6450216Z         	            	 (string) (len=12) "-force=false",
2022-06-02T00:53:12.6450744Z         	            	 (string) (len=14) "test_keyspace3"
2022-06-02T00:53:12.6451039Z         	            	}
2022-06-02T00:53:12.6451319Z         	Test:       	TestCreateKs
```

**Comment by notfelineit** [comment]
@[arvind-murty](https://github.com/arvind-murty) did you run `make web_build`? That might help compile!

**Comment by notfelineit** [comment]
Hey @arvind-murty, I went ahead and created a branch that removes those fields from vtctld2, you can rebase off https://github.com/vitessio/vitess/pull/10459 to get those changes onto your branch!

File: go/cmd/vtctldclient/command/keyspaces.go [hunk 1]
```diff
@@ -503,8 +449,6 @@ func commandValidateVersionKeyspace(cmd *cobra.Command, args []string) error {
 func init() {
 	CreateKeyspace.Flags().BoolVarP(&createKeyspaceOptions.Force, "force", "f", false, "Proceeds even if the keyspace already exists. Does not overwrite the existing keyspace record")
 	CreateKeyspace.Flags().BoolVarP(&createKeyspaceOptions.AllowEmptyVSchema, "allow-empty-vschema", "e", false, "Allows a new keyspace to have no vschema")
-	CreateKeyspace.Flags().StringVar(&createKeyspaceOptions.ShardingColumnName, "sharding-column-name", "", "The column name to use for sharding operations")
-	CreateKeyspace.Flags().Var(&createKeyspaceOptions.ShardingColumnType, "sharding-column-type", "The type of the column to use for sharding operations")
```
Comment (deepthi) [inline go/cmd/vtctldclient/command/keyspaces.go:507]: @ajm188 what is the right way to deprecate these options?

File: go/vt/vtctl/vtctl.go [hunk 2]
```diff
@@ -1919,8 +1919,6 @@ func commandDeleteShard(ctx context.Context, wr *wrangler.Wrangler, subFlags *fl
 }
 
 func commandCreateKeyspace(ctx context.Context, wr *wrangler.Wrangler, subFlags *flag.FlagSet, args []string) error {
-	shardingColumnName := subFlags.String("sharding_column_name", "", "Specifies the column to use for sharding operations")
-	shardingColumnType := subFlags.String("sharding_column_type", "", "Specifies the type of the column to use for sharding operations")
```
Comment (deepthi) [inline go/vt/vtctl/vtctl.go:1923]: @ajm188 what is the right way to deprecate these?

**Comment by arvind-murty** [comment]
> Hey @arvind-murty, I went ahead and created a branch that removes those fields from vtctld2, you can rebase off #10459 to get those changes onto your branch!

Thank you. Since your PR was merged, I just did a merge from main.

File: go/cmd/vtctldclient/command/keyspaces.go (hunk 1 above)
Comment (ajm188) [reply go/cmd/vtctldclient/command/keyspaces.go:511]: [`pflag` has native support for this](https://pkg.go.dev/github.com/spf13/pflag#readme-deprecating-a-flag-or-its-shorthand)!! (_vaguely gestures at "won't it be great to use `pflag` everywhere in the future?"_):

```go
CreateKeyspace.Flags().StringVar(&createKeyspaceOptions.ShardingColumnName, "sharding-column-name", ...)
CreateKeyspace.Flags().MarkDeprecated("sharding-column-name", "Keyspace.ShardingColumnName is no longer used for resharding") // feel free to tweak the deprecation message, i didn't give this much thought
// same pattern for ShardingColumnType
```

Comment (deepthi) [reply go/cmd/vtctldclient/command/keyspaces.go:511]: Thank you. I'm thinking we should let this PR delete these flags on main (to go into v15) and do a separate PR to deprecate them in v14 (on release-14.0)

Comment (ajm188) [reply go/cmd/vtctldclient/command/keyspaces.go:511]: makes sense to me!

**Comment by ajm188** [comment]
> I'm in two minds about whether we need to add this to the summary doc. v2 resharding (not the same as vreplication v2) is old enough that probably no one cares.
> We _are_ announcing the deprecation of these fields for the CLI commands in v14.
> @ajm188 @frouioui do you have any opinions on this?

It's been marked as ["legacy"](https://vitess.io/docs/13.0/user-guides/historical/) since at least v13, so I think it's fine to just delete. If we feel a little iffy about it, we can include a "minor deletions/deprecations" section or something like that towards the bottom of the doc.

**Pull request merged**